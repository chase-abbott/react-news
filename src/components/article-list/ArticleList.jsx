import React from 'react';
import PropTypes from 'prop-types';
import Article from '../article/Article.jsx';

const ArticleList = ({ articles }) => {
  const articleItems = articles.map(article => (
    <li key={article.title}>
      <Article article={article}/>
    </li>
  ));
  return <ul> {articleItems}</ul>;
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired
    })
  )
};

export default ArticleList;
