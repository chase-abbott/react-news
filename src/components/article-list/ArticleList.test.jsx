import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList.jsx';

describe('Article componenet', () => {
  it('renders a title, author, and description', () => {
    const articles = [{
      title: 'Sports team wins',
      author: 'Chase Abbott',
      description: 'They won by a lot'
    },
    {
      title: 'Dog learns spanish',
      author: 'Nelly the Pomeranian',
      description: 'A dog that bairly knew english learns spanish'
    }];

    const { asFragment } = render(<ArticleList articles={articles}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
