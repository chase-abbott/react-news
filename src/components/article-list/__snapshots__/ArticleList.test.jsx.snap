// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article componenet renders a title, author, and description 1`] = `
<DocumentFragment>
  <ul>
     
    <li>
      <div>
        <p>
           Sports team wins
        </p>
        <p>
           Chase Abbott
        </p>
        <span>
           They won by a lot
        </span>
      </div>
    </li>
    <li>
      <div>
        <p>
           Dog learns spanish
        </p>
        <p>
           Nelly the Pomeranian
        </p>
        <span>
           A dog that bairly knew english learns spanish
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
